src/layouts/EditHomepage.jsx

```diff
@@ -1171,14 +1173,36 @@ const EditHomepage = ({ match }) => {
   return (
     <>
       {itemPendingForDelete.id && (
-        <DeleteWarningModal
-          onCancel={() => setItemPendingForDelete({ id: null, type: "" })}
-          onDelete={() => {
-            deleteHandler(itemPendingForDelete.id)
+        <GenericWarningModal
+          isOpen={isOpen}
+          onClose={() => {
             setItemPendingForDelete({ id: null, type: "" })
+            onClose()
           }}
-          type={itemPendingForDelete.type}
-        />
+          displayTitle={`Delete ${itemPendingForDelete.type} section`}
+          displayText={`Are you sure you want to delete ${itemPendingForDelete.type}?`}
+        >
+          <Button
+            variant="ghost"
+            colorScheme="secondary"
+            onClick={() => {
+              setItemPendingForDelete({ id: null, type: "" })
```

Comment:
we seem to persistently repeat this pattern of having a bound state, invoking a function with teh bound state or otherwise resetting.

we should aim to create a hook to abstract over this functionality then invoking it only on `GenericWarningModal`'s delete confirmation

Reply:
As discussed offline - this will not be the desired final behaviour, and due to the complexity of editHomepage, we are electing to leave this for a future refactor rather than implementing a temporary solution